orion/controllers/programs/programs_test.go

```diff
@@ -11,9 +11,78 @@ func TestExample(t *testing.T) {
   }
 }
 
-func TestCheckProgram(t *testing.T) {
-  isValid := CheckValidProgram()
-  if (isValid == false) {
-    t.Errorf("Check was incorrect, got: %t, expected: %t.", isValid, false)
+func TestValidProgramId(t *testing.T) {
+  program := Program{ProgramId: "test_program", Name: "Test Program", Grade1: 1, Grade2: 12, Description: "Description"}
+  err := CheckValidProgram(program)
+  if err != nil {
+    t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+  }
+
+  program = Program{ProgramId: "a_a_a_a_a_a", Name: "Test Program", Grade1: 1, Grade2: 12, Description: "Description"}
+  err = CheckValidProgram(program)
+  if err != nil {
+    t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+  }
+
+  program = Program{ProgramId: "This_8932792_352_IS_3589_A_vaLiD_pRogRAm_iD_6A783B3S", Name: "Test Program", Grade1: 1, Grade2: 12, Description: "Description"}
+  err = CheckValidProgram(program)
+  if err != nil {
+    t.Errorf("Check was incorrect, got: %s, expected: nil", err.Error())
+  }
+
+  program = Program{ProgramId: "a__a", Name: "Test Program", Grade1: 1, Grade2: 12, Description: "Description"}
+  err = CheckValidProgram(program)
+  if err == nil {
+    t.Errorf("Check was incorrect, got: nil, expected: invalid program id")
+  }
+
+  program = Program{ProgramId: "_", Name: "Test Program", Grade1: 1, Grade2: 12, Description: "Description"}
+  err = CheckValidProgram(program)
+  if err == nil {
+    t.Errorf("Check was incorrect, got: nil, expected: invalid program id")
+  }
+
+  program = Program{ProgramId: "", Name: "Test Program", Grade1: 1, Grade2: 12, Description: "Description"}
+  err = CheckValidProgram(program)
+  if err == nil {
+    t.Errorf("Check was incorrect, got: nil, expected: invalid program id")
   }
 }
+
+func TestValidGrades(t *testing.T) {
+  program := Program{ProgramId: "test_program", Name: "Test Program", Grade1: 1, Grade2: 12, Description: "Description"}
```

Comment:
It seems like we are creating a new Program object for every subtest.
How about at the beginning of the test we create a "base Program", and then for every subtest, we only change the `grade1` and `grade2` fields?

So you can have:
```
program := Program{...}
...

program.Grade1 = 2
program.Grade2 = 3
...
program.Grade1 = 12
program.Grade2 = 0
```
Otherwise, lots of code redundancy :)